docs/proposals/caching.adoc

```diff
@@ -0,0 +1,105 @@
+
+2. Cache prior build output to support build skipping (see proposal at https://github.com/opendevstack/ods-pipeline/pull/423). Usage of the global cache for this purpose should work analog to how this is done for dependency caching but it is not further elaborated on here as this is covered in the mentioned proposal.
+
+3. There may be other usages for example for the gradle wrapper that could perhaps benefit from a global cache.
+
+The following new parameter is introduced to build tasks supporting dependency caching:
+
+* `cache-deps` a boolean defaulting to `false`. Only if set to `true` the task will used dependency caching. 
+
+In addition to the build task parameter, build tasks also receives the following:
+
+* File `.ods/cache-deps-parent-dir` contains an absolute path without trailing '/' to an existing directory (no whitespace or newlines). If file `.ods/cache-deps-parent-dir` does not exits the task must not use dependency caching. This is used to switch off dependency caching dynamically for example with a special tag in a git commit message. 
+
+Cleanup will spare directories below the $(cache-deps-parent-dir) so that build tasks must keep their cached files in a subdirectory they create. For example in `$(cache-deps-parent-dir)/go-mods/` where `go-mods` is called technology-name in this proposal.
+
+Build tasks supporting caching will also be adjusted to
+
+- Log available disk space at the beginning of the build.
+
+- Log how long build commands which may be long running take in seconds. 
+
+- Log available disk space at the end of the build and also the space delta.
+
+
+=== ods-start and cleanup
+
+Build tasks supporting caching can count on:
+
+* cache cleanups not occurring while they are running
+* cleanups will not be partially completed. 
+
+On the other hand build tasks must not assume that the cache is still available from a prior build. 
+
+The dependency cache is somewhat special is it typically does not make sense to remove individual files.
+
+Instead if disk space issues arise one should manually:
+
+* Increase the PVC space of the associated repository or
+* Recreate the PVC (TODO: should this be done automatically for example on a randomized bi-weekly schedule for example)
+
+However in general cleanup of the PVC does happen during `ods-start`.
+
+The cleanup strategy described here is subject to change even in patch versions and specific details must not be relied on.
+
+The following cache locations on the PVC are used:
+
+- `+/.c/deps/<technology-name>/+` for dependency caches of a particular build technology. The technology-name would be defined by the build script and unknown to `ods-start`. 
```

Comment:
I'd prefer e.g. `/.ods-cache/dependencies/<technology-name>/`

Reply:
I find this a bit lengthy. How about `. /.ods-cache/deps/<technology-name/` ?